slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -122,8 +119,9 @@ trait MySQLProfile extends JdbcProfile { profile =>
       sym.flatMap(_.findColumnOption[RelationalProfile.ColumnOption.Length]) match {
         case Some(l) => if(l.varying){
             //http://dev.mysql.com/doc/refman/5.7/en/string-type-overview.html
-            if(l.length <= 65535 ) s"VARCHAR(${l.length})" 
-            else if(l.length <= 16777215 ) "MEDIUMTEXT"
+            if(l.length < 65535) s"VARCHAR(${l.length})"
+            else if(l.length == 65535) "TEXT"
```

Comment:
mysql jdbc driver sets type TEXT to length 65535. Setting TEXT for length 65535 is safer since there is a row size limit .

```bash
[error] Test slick.test.codegen.GeneratedCodeTest.allTests failed: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Row size too large. The maximum row size for the used table type, not counting BLOBs, is 65535. This includes storage overhead, check the manual. You have to change some columns to TEXT or BLOBs, took 51.986 sec

```